.script/SolutionValidations/validSupportObject.ts

```diff
@@ -0,0 +1,79 @@
+import { MainTemplateSupportObjectValidationError } from "./../utils/validationError";
+import { ExitCode } from "../utils/exitCode";
+import fs from "fs";
+
+// function to check if the solution has a valid support object
+export function IsValidSupportObject(filePath: string): ExitCode {
+
+    // check if the file is a mainTemplate.json file
+    if (filePath.endsWith("mainTemplate.json")) {
+        // read the content of the file
+        let jsonFile = JSON.parse(fs.readFileSync(filePath, "utf8"));
+
+        // check if the file has a "resources" field
+        if (!jsonFile.hasOwnProperty("resources")) {
+            console.warn(`No "resources" field found in the file. Skipping file path: ${filePath}`);
```

Comment:
Shouldn't be this error? Is there any scenario where you don't have resources in main template?

Reply:
Added the exception when there is no resource exists.